parsing/location.mli

```diff
@@ -104,9 +107,6 @@ type error =
 
 exception Error of error
 
-val print_error_prefix: formatter -> unit -> unit
```

Comment:
I removed the only use of this (in the expect test tool), so it seems appropriate to stop exposing it too.